app/web/src/components/ModelingDiagram/ModelingDiagram.vue

```diff
@@ -3160,35 +3171,63 @@ function fixRenameInputPosition() {
       ];
     if (componentBox && renameInputWrapperRef.value) {
       const { x, y } = convertGridCoordsToContainerCoords(componentBox);
+      const z = zoomLevel.value;
 
       if (renameElement.value instanceof DiagramNodeData) {
         // moving the input box for a Node
-        renameInputWrapperRef.value.style.top = `${y + 3}px`;
-        renameInputWrapperRef.value.style.left = `${
-          x + 8 - componentBox.width / 2
-        }px`;
-        renameInputWrapperRef.value.style.width = `${
-          componentBox.width - NODE_TITLE_HEADER_MARGIN_RIGHT
-        }px`;
+        const top = y + 3 * z;
+        const left =
+          z > 0.5
+            ? x + 8 * z - (componentBox.width * z) / 2
+            : x - (componentBox.width * z) / 2;
+        const width =
+          z > 0.5
+            ? (componentBox.width - NODE_TITLE_HEADER_MARGIN_RIGHT) * z
+            : componentBox.width * z;
```

Comment:
this math makes me cry a little, but i can understand why zoom level makes things tricky...